pkg/standalone/standalone.go

```diff
@@ -612,41 +612,45 @@ func untar(filepath, targetDir, binaryFilePrefix string) (string, error) {
 
 	tr := tar.NewReader(gzr)
 
+	foundBinary := ""
 	for {
 		header, err := tr.Next()
 
-		switch {
-		case err == io.EOF:
-			return "", fmt.Errorf("file is empty")
-		case err != nil:
+		if err == io.EOF {
+			break
+		} else if err != nil {
 			return "", err
-		case header == nil:
+		} else if header == nil {
 			continue
 		}
 
-		extractedFilePath := path.Join(targetDir, header.Name)
-
-		switch header.Typeflag {
-		case tar.TypeReg:
-			// Extract only the binaryFile
-			if header.Name != binaryFilePrefix {
-				continue
-			}
-
-			f, err := os.OpenFile(extractedFilePath, os.O_CREATE|os.O_RDWR, os.FileMode(header.Mode))
-			if err != nil {
+		// untar all files in archive
+		path := path_filepath.Join(targetDir, header.Name)
+		info := header.FileInfo()
+		if info.IsDir() {
+			if err = os.MkdirAll(path, info.Mode()); err != nil {
 				return "", err
 			}
+			continue
+		}
 
-			// #nosec G110
-			if _, err := io.Copy(f, tr); err != nil {
-				return "", err
-			}
-			f.Close()
+		f, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, info.Mode())
```

Comment:
these permissions are different than earlier change,,os.O_RDWR missing..? Do we need this additional permissions ?